src/main/kotlin/org/ptss/support/infrastructure/persistence/entities/GroupEntity.kt

```diff
@@ -0,0 +1,64 @@
+package org.ptss.support.infrastructure.persistence.entities
+
+import jakarta.persistence.*
+import org.hibernate.annotations.Check
+import org.hibernate.annotations.OnDelete
+import org.hibernate.annotations.OnDeleteAction
+
+@Entity
+@Table(
+    name = "groups",
+    uniqueConstraints = [
+        UniqueConstraint(columnNames = ["patient_id"]),
+        UniqueConstraint(columnNames = ["primary_caregiver_id"])
+    ]
+)
+// Ensures patient, healthcare professional, and caregiver IDs are distinct in the database
+@Check(
+    constraints = """
+        patient_id != healthcare_professional_id AND 
+        patient_id != primary_caregiver_id AND 
+        healthcare_professional_id != primary_caregiver_id
+    """
+)
+class GroupEntity : BaseEntity() {
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(nullable = false, updatable = false)
+    @OnDelete(action = OnDeleteAction.CASCADE)  // When patient is deleted, delete the group
+    var patient: UserEntity? = null
```

Comment:
nullable is set to false, but is active is determined by the patient not being null?
    val isActive: Boolean
        get() = patient != null

Reply:
Goeie, thanks :octocat: